Realm/Realm/Configurations/SyncConfigurationBase.cs

```diff
@@ -133,11 +134,18 @@ internal override async Task<SharedRealmHandle> CreateHandleAsync(Schema.RealmSc
 
         internal virtual Native.SyncConfiguration CreateNativeSyncConfiguration()
         {
-            var clientResyncMode = ClientResyncMode.DiscardLocal;
+            var clientResyncMode = ClientResyncMode.AutomaticRecoveryOrDiscardLocal;
 
-            if (ClientResetHandler != null && ClientResetHandler is ManualRecoveryHandler)
+            if (ClientResetHandler != null)
             {
-                clientResyncMode = ClientResyncMode.Manual;
+                clientResyncMode = ClientResetHandler switch
```

Comment:
Instead of switching on type, can we add an abstract internal property to the ClientResetHandlerBase class and have inheritors be explicit about the resync mode they support.

Reply:
Beside allowing a param to the constructor of `AutomaticReccoveryHandler`, as you suggested [here](https://github.com/realm/realm-dotnet/pull/2956#discussion_r899977885). Is there any other advantage to this approach? E.g. do you see handlers ever changing internal behaviour after being released? Also reading a property is surely faster than switching on the type, but were you concerned with performance?

Reply:
All I'm trying to say is, the type of the recovery strategy already implies what the recovery behaviour is gonna be. Why do we need to write it "again" in the class?